src/olympia/addons/models.py

```diff
@@ -1307,16 +1307,30 @@ def is_recommended(self):
                 addon=self).exists()
         return recommended
 
-    @cached_property
-    def promoted_group(self):
-        """If the addon is promoted then return the Promoted group details."""
+    def is_promoted(self, group=None, application=None, currently=True):
+        """Is the addon currently promoted for the specified group and
+        application.
+
+        `group` is the PromotedClass class; if group=None (default) then all
+        promotion groups are considered;
+        `application` is the App class; if application=None (default) then all
+        applications are considered;
+        `currently=True` means only returns True if self.current_version is
```

Comment:
I find this argument name not very self-explanatory.

Reply:
I'm open to better naming.

Reply:
`currently_approved` is a suggestion